src/include/duckdb/common/types/string.hpp

```diff
@@ -0,0 +1,190 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/types/string.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/numeric_utils.hpp"
+#include "duckdb/common/typedefs.hpp"
+
+namespace duckdb {
+class String {
+public:
```

Comment:
I think  we need also String(String) constructor + move constructor + copy / move assignment, the default one are not correct here.

Reply:
Yeah, we should also consider if copy actually copies the underlying data (so it's owning) or creates a non-owning reference. Perhaps we should just disable copy construction completely and make explicit `.Copy()` and  `.Reference()`/`.Clone()` methods to specify the behavior.